### DISCUSSION TITLE
choosing the best controller for a project

### DISCUSSION THREAD

**Discussion by SanjayEIC** [Moddable-OpenSource/moddable#1306] [Q&A]
**Purpose**: We have a diagnostic module, and we need to decide which controller to use, either ESP32-S3 or ESP32-U4. we 
                     used moddable lib.

**Questions: -** 
1) What is the lifespan or support duration for each controller, and what is the availability of controllers in the market?
2) Are both controllers currently under development?
3) Are there any major or minor challenges or issues associated with both controllers?

**Comment by phoddie** [comment]
If I understand the questions correctly, Espressif would be best able answer.

I am curious: what does your diagnostic module do?

**Reply by SanjayEIC** [reply]
Diagnostic module for debugging purposes and firmware update. through customized protocol  
**this is the flow** 

mobile <--------------> **diagnostics module** <-----(physical wire) --------> appliance /device  

service module is connected to mobile via BLE or SoftAP(wifi) and transmits data to service module. and service module process and forward data to device.  through physical cable.

**Reply by phoddie** [reply]
That's interesting. Seems pretty straightforward.

You probably shoudl check with Espressif on their long-term support plans for the silicon options you are considering. All things being equal, the ESP32-S3 is more recent and has some advantages, so that would seem to be the better choice. If cost is a significant issue (seems unlikely here), perhaps the ESP32-U4WDH is worth considering.

**Comment by SanjayEIC** [comment]
Ok thanks for the support.
